### PR TITLE
table partition: exchange partition telemetry

### LINKED ISSUE #37874: table partition: add telemetry for exchange partition
## Enhancement

Add telemetry for exchange partition.

### PR DESCRIPTION
**Pull request by ymkzpx**
### What problem does this PR solve?

Issue Number: close #37874

Problem Summary:

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None.
```

### CODE REVIEW

File: telemetry/data_feature_usage_test.go [hunk 1]
```diff
@@ -241,6 +241,17 @@ func TestTablePartition(t *testing.T) {
...
+	tk.MustExec("drop table if exists pt2")
+	tk.MustExec("create table pt2 (a int,b int) partition by range(a) (" +
+		"partition p0 values less than (10)," +
+		"partition p1 values less than (20))")
+	tk.MustExec("drop table if exists nt")
+	tk.MustExec("create table nt (a int,b int)")
+	tk.MustExec(`alter table pt2 exchange partition p1 with table nt`)
+	usage, err = telemetry.GetFeatureUsage(tk.Session())
+	require.NoError(t, err)
+	require.Equal(t, int64(1), usage.ExchangePartition.ExchangePartitionCnt)
```
Comment (mjonss) [inline telemetry/data_feature_usage_test.go:254]: Please also add a test above to verify that it was zero before this part of the test.

**Review by mjonss — APPROVED** [review]
LGTM, with one small suggestion to verify that the start value in the test is zero.

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**